opennebula_api/models.py

```diff
@@ -208,6 +217,33 @@ def get_vm(self, vm_id):
         except:
             raise ConnectionRefusedError
 
+    def get_primary_ipv4(self, vm_id):
+        """
+        Returns the primary IPv4 of the given vm.
+        To be changed later.
+
+        :return: An IP address string, if it exists else returns None
+        """
+        all_ipv4s = self.get_vm_ipv4_addresses(vm_id)
+        if len(all_ipv4s) > 0:
+            return all_ipv4s[0]
```

Comment:
I suggest to even return the whole list as in " ".join(all_ipv4s)

Reply:
We have `get_vm_ipv4_addresses` below for that purpose @telmich. I created the above method to identify which IP address to use to run the cdist config on.